source/adapters/level_zero/adapter.cpp

```diff
@@ -171,8 +209,40 @@ ur_adapter_handle_t_::ur_adapter_handle_t_()
 
       return;
     }
+    // Dynamically load the new L0 SysMan separate init and new EXP apis
+    // separately. This must be done to avoid attempting to use symbols that do
+    // not exist in older loader runtimes.
+#ifdef _WIN32
+    std::string l0LoaderName = "ze_loader.dll";
+#else
+    std::string l0LoaderName = "libze_loader.so.1";
+#endif
+    GlobalAdapter->loaderHandle =
+        ur_loader::LibLoader::loadAdapterLibrary(l0LoaderName.c_str());
+    if (GlobalAdapter->loaderHandle.get() != nullptr) {
+      GlobalAdapter->getDeviceByUUIdFunctionPtr =
+          (zes_pfnDriverGetDeviceByUuidExp_t)
+              ur_loader::LibLoader::getFunctionPtr(
+                  GlobalAdapter->loaderHandle.get(),
+                  "zesDriverGetDeviceByUuidExp");
+      GlobalAdapter->getSysManDriversFunctionPtr =
+          (zes_pfnDriverGet_t)ur_loader::LibLoader::getFunctionPtr(
+              GlobalAdapter->loaderHandle.get(), "zesDriverGet");
+      GlobalAdapter->sysManInitFunctionPtr =
+          (zes_pfnInit_t)ur_loader::LibLoader::getFunctionPtr(
+              GlobalAdapter->loaderHandle.get(), "zesInit");
+    }
+    if (GlobalAdapter->getDeviceByUUIdFunctionPtr &&
+        GlobalAdapter->getSysManDriversFunctionPtr &&
+        GlobalAdapter->sysManInitFunctionPtr) {
+      logger::debug("\nzesInit with flags value of {}\n", static_cast<int>(0));
```

Comment:
This will always print value of 0. Was that intentional?

Reply:
It was, but let me create a variable holding the current flag value instead of making this fixed.

Reply:
Added a variable to allow this value to change.